### ISSUE TITLE
Question: How to count current translations/transcodings

### ISSUE THREAD

**Issue by boettner** [sipwise/rtpengine#908]
Hi folks, is it possible to get statistics about current translated or transcoded streams from rtpengine-ctl? I would like to count/graph the amount of transcoded G.729 -> G.711 calls. And it would be nice to count the translations from SRTP->RTP and vice versa.

**Comment by rfuchs** [comment]
Not as such directly (would be NTH), but you can pull the full list of calls, iterate through them in a script, pull the details for each call, and compare protocols/codecs used on both sides.

**Comment by boettner** [comment]
Thanks a lot for your quick response.

I tried to get information about a translated call, but wasn't able to detect protocol changes. This call was translated from SRTP -> RTP:

```
rtpengine-ctl list sessions 313537393037393437313232303933-ngx8winshzhj

callid:                  313537393037393437313232303933-ngx8winshzhj | deletionmark: yes | created:  1579079473  | proxy:10.0.0.19:52336 | tos:184 | last_signal:1579079473 | redis_keyspace:0 | foreign:no

--- Tag 'hw6dxh1brp' type: FROM_TAG, callduration 24.472151 , in dialogue with 'as157a3bcd'
------ Media #1, 10.0.0.233:15556 <>   10.1.0.72:63402, 1195 p, 215702 b, 0 e, 1579079499 last_packet
------ Media #1, 10.0.0.233:15557 <>   10.1.0.72:63403 (RTCP), 7 p, 432 b, 0 e, 1579079498 last_packet
------ Media #2,         0.0.0.0:0     <>   10.1.0.72:63402, 0 p, 0 b, 0 e, 1579079473 last_packet
------ Media #2,         0.0.0.0:0     <>   10.1.0.72:63403 (RTCP), 0 p, 0 b, 0 e, 1579079473 last_packet
--- Tag 'as157a3bcd' type: TO_TAG, callduration 24.272797 , in dialogue with 'hw6dxh1brp'
------ Media #1, 10.0.0.108:15580 <> 10.0.0.100:20946, 1191 p, 204852 b, 0 e, 1579079499 last_packet
------ Media #1, 10.0.0.108:15581 <> 10.0.0.100:20947 (RTCP), 4 p, 256 b, 0 e, 1579079494 last_packet
------ Media #2,         0.0.0.0:0     <> 10.0.0.100:0    , 0 p, 0 b, 0 e, 1579079473 last_packet
------ Media #2,         0.0.0.0:0     <> 10.0.0.100:1     (RTCP), 0 p, 0 b, 0 e, 1579079473 last_packet
```
(This might look different for codec translations.)

**Comment by rfuchs** [comment]
Codec information is available in master (8322a8b3)
I guess protocol info is actually missing. I'll add it.

**Comment by boettner** [comment]
Thank you very much. I'll test it when available.
